### PR TITLE
refactor: setup project for TypeScript

### PR DESCRIPTION
**Pull request by EscapedGibbon**

### CODE REVIEW

**Comment by stropitek** [comment]
Please fix https://github.com/mljs/rolling-ball-baseline/issues/19 in this PR

File: src/__tests__/test.rollingBall.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ describe('test rollingball', () => {
       ).toBeLessThan(0.2);
     }
 
-    let y = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1];
+    let y = new Float64Array([1, 1, 1, 1, 1, 1, 1, 1, 1, 1]);
```
Comment (stropitek) [inline src/__tests__/test.rollingBall.ts]: Please revert this, you should not have to change the tests

File: src/rollingBall.ts [hunk 2]
```diff
@@ -32,7 +40,7 @@ export function rollingBall(spectrum, options = {}) {
     throw new TypeError('Spectrum must not be empty');
   }
 
-  const numberPoints = spectrum.length;
+  const numberPoints: number = spectrum.length;
```
Comment (stropitek) [inline src/rollingBall.ts]: You don't need to add a type annotation here. TS knows that spectrum is an array, so it knows that it has a property `length` which is a number.  When possible, always let TS infer the types for you.

File: src/rollingBall.ts [hunk 3]
```diff
@@ -23,7 +23,15 @@ import { xMean, xMaxValue, xMinValue } from 'ml-spectra-processing';
...
  * @param {Number} [options.windowS] - width of local window for smoothing, defaults to 8% of the spectrum length
  */
-export function rollingBall(spectrum, options = {}) {
+
+interface Options {
+  windowM: number;
+  windowS: number;
+}
+export function rollingBall(
+  spectrum: Float64Array,
```
Comment (stropitek) [inline src/rollingBall.ts]: Please use the `NumberArray` utility type from the `cheminfo-types` package.

File: testData/spectrum.ts [hunk 4]
```diff
@@ -1,4 +1,4 @@
-export const spectrum = [
+export const spectrum = new Float64Array([
```
Comment (stropitek) [inline testData/spectrum.ts]: Please revert this change

File: src/rollingBall.ts [hunk 5]
```diff
@@ -46,8 +54,8 @@ export function rollingBall(spectrum, options = {}) {
 
   // fi(1) in original paper
   for (let i = 0; i < spectrum.length; i++) {
-    let windowLeft = Math.max(0, i - windowM);
-    let windowRight = Math.min(i + windowM + 1, spectrum.length);
+    let windowLeft: number = Math.max(0, i - windowM);
```
Comment (stropitek) [inline src/rollingBall.ts]: ditto

File: src/rollingBall.ts [hunk 6]
```diff
@@ -46,8 +54,8 @@ export function rollingBall(spectrum, options = {}) {
...
+    let windowRight: number = Math.min(i + windowM + 1, spectrum.length);
```
Comment (stropitek) [inline src/rollingBall.ts]: ditto

File: src/rollingBall.ts [hunk 7]
```diff
@@ -57,17 +65,17 @@ export function rollingBall(spectrum, options = {}) {
 
   // fi in original paper
   for (let i = 0; i < minima.length; i++) {
-    let windowLeft = Math.max(0, i - windowM);
-    let windowRight = Math.min(i + windowM + 1, minima.length);
+    let windowLeft: number = Math.max(0, i - windowM);
```
Comment (stropitek) [inline src/rollingBall.ts]: ditto

File: src/rollingBall.ts [hunk 8]
```diff
@@ -23,7 +24,15 @@ import { xMean, xMaxValue, xMinValue } from 'ml-spectra-processing';
...
  */
-export function rollingBall(spectrum, options = {}) {
+
+interface Options {
+  windowM: number;
+  windowS: number;
+}
+export function rollingBall(
+  spectrum: NumberArray,
+  options: Partial<Options> = {}, // not specific enough => put two optional parameters with specific type
```
Comment (stropitek) [inline src/rollingBall.ts]: What do you mean with the comment?

Comment (EscapedGibbon) [reply src/rollingBall.ts]: My bad. It doesn't mean anything now. just forgot to remove it.

**Review by stropitek — APPROVED** [review]
